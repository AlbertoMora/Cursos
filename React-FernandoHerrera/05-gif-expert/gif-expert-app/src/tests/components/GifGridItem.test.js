import React from "react";
import { shallow } from "enzyme";

import GifGridItem from "../../components/GifGridItem";

describe("Render testing of GifGrid component", () => {
    const title = "Title";
    const img = "http://localhost:3000/favicon.ico";
    const wrapper = shallow(<GifGridItem title={title} url={img} />);

    test("should render the component correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
    test("should have a <p></p> with the title", () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);
    });
    test("should have the correct url and props attributes", () => {
        const image = wrapper.find("img");

        expect(image.prop("src")).toBe(img);
        expect(image.prop("alt")).toBe(title);
    });
    test("should have the className animate__backInLeft", () => {
        const expectedClass = "animate__backInLeft";
        const div = wrapper.find("div");

        expect(div.prop("className")).toContain(expectedClass);
    });
});
