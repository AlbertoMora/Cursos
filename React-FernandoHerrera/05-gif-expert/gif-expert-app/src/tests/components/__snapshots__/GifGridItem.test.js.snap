// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render testing of GifGrid component should render the component correctly 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="Title"
    src="http://localhost:3000/favicon.ico"
  />
  <p>
    Title
  </p>
</div>
`;
